### PR TITLE
Lint-roller PR

### PR DESCRIPTION
**Pull request by TacoTheDank**
Here it is. Hope you like it. It's a bit large, sorry.

### CODE REVIEW

File: Kuroba/app/src/main/java/com/github/k1rakishou/chan/core/manager/ReplyManager.java [hunk 1]
```diff
@@ -53,7 +53,7 @@ public void putReply(ChanDescriptor chanDescriptor, Reply reply) {
         // Remove files from all other replies because there can only be one picked_file at the same time.
         // Not doing this would be confusing and cause invalid fileNames.
         for (Map.Entry<ChanDescriptor, Reply> entry : drafts.entrySet()) {
-            if (!entry.getKey().equals(chanDescriptor)) {
+            if (entry.getKey() != chanDescriptor) {
```
Comment (K1rakishou) [inline Kuroba/app/src/main/java/com/github/k1rakishou/chan/core/manager/ReplyManager.java]: Here should be "equals" not "!=" because this is java, not kotlin.

File: Kuroba/app/src/main/java/com/github/k1rakishou/chan/core/site/parser/CommentParser.java [hunk 2]
```diff
@@ -334,7 +334,7 @@ private boolean isPostLinkableAlreadyAdded(SpannableString res, PostLinkable.Val
         }
 
         for (PostLinkable postLinkable : alreadySetPostLinkables) {
-            if (linkValue.equals(postLinkable.getLinkableValue())) {
+            if (linkValue == postLinkable.getLinkableValue()) {
```
Comment (K1rakishou) [inline Kuroba/app/src/main/java/com/github/k1rakishou/chan/core/site/parser/CommentParser.java]: Here should be "equals" too.

File: Kuroba/app/src/main/java/com/github/k1rakishou/chan/features/settings/SettingsCoordinator.kt [hunk 3]
```diff
@@ -253,29 +253,21 @@ class SettingsCoordinator(
...
+    val currentScreen = (if (screenStack.isEmpty()) {
       null
     } else {
       screenStack.peek()
-    }
-
-    if (currentScreen == null) {
-      return null
-    }
+    }) ?: return null
```
Comment (K1rakishou) [inline Kuroba/app/src/main/java/com/github/k1rakishou/chan/features/settings/SettingsCoordinator.kt]: No, please, don't. This is really hard to read.

File: Kuroba/app/src/main/java/com/github/k1rakishou/chan/features/settings/SettingsCoordinator.kt [hunk 4]
```diff
@@ -253,29 +253,21 @@ class SettingsCoordinator(
...
+    val currentScreen = (if (screenStack.isEmpty()) {
       null
     } else {
       screenStack.peek()
-    }
-
-    if (currentScreen == null) {
-      return
-    }
+    }) ?: return
```
Comment (K1rakishou) [inline Kuroba/app/src/main/java/com/github/k1rakishou/chan/features/settings/SettingsCoordinator.kt]: Same here.

File: Kuroba/app/src/main/java/com/github/k1rakishou/chan/ui/controller/LogsController.java [hunk 5]
```diff
@@ -109,11 +109,11 @@ public static String loadLogs(int linesCount) {
 
         InputStream outputStream = process.getInputStream();
         //This filters our log output to just stuff we care about in-app (and if a crash happens, the uncaught handler gets it and this will still allow it through)
-        String filtered = "";
+        StringBuilder filtered = new StringBuilder();
         for (String line : IOUtils.readString(outputStream).split("\n")) {
-            if (line.contains(getApplicationLabel())) filtered = filtered.concat(line).concat("\n");
+            if (line.contains(getApplicationLabel())) filtered = new StringBuilder((filtered + line) + "\n");
```
Comment (K1rakishou) [inline Kuroba/app/src/main/java/com/github/k1rakishou/chan/ui/controller/LogsController.java]: I think it's better to just use "append()" here instead of creating a new StringBuilder() on every iteration.

File: Kuroba/app/src/main/java/com/github/k1rakishou/chan/ui/view/PostImageThumbnailView.java [hunk 6]
```diff
@@ -162,9 +162,7 @@ private String getUrl(PostImage postImage) {
         boolean hasImageUrl = postImage.imageUrl != null;
 
         if (highRes && hasImageUrl && (!postImage.spoiler() || ChanSettings.removeImageSpoilers.get())) {
-            url = postImage.type == ChanPostImageType.STATIC
-                    ? postImage.imageUrl.toString()
-                    : postImage.getThumbnailUrl().toString();
+            url = (postImage.type == ChanPostImageType.STATIC ? postImage.imageUrl : postImage.getThumbnailUrl()).toString();
```
Comment (K1rakishou) [inline Kuroba/app/src/main/java/com/github/k1rakishou/chan/ui/view/PostImageThumbnailView.java]: I don't think this makes it more readable.

**Review by K1rakishou — CHANGES_REQUESTED** [review]

File: Kuroba/app/src/main/java/com/github/k1rakishou/chan/core/manager/ReplyManager.java (hunk 1 above)
Comment (TacoTheDank) [reply Kuroba/app/src/main/java/com/github/k1rakishou/chan/core/manager/ReplyManager.java]: But "!=" exists in Java...?

File: Kuroba/app/src/main/java/com/github/k1rakishou/chan/core/site/parser/CommentParser.java (hunk 2 above)
Comment (TacoTheDank) [reply Kuroba/app/src/main/java/com/github/k1rakishou/chan/core/site/parser/CommentParser.java]: See above.

File: Kuroba/app/src/main/java/com/github/k1rakishou/chan/core/manager/ReplyManager.java (hunk 1 above)
Comment (TacoTheDank) [reply Kuroba/app/src/main/java/com/github/k1rakishou/chan/core/manager/ReplyManager.java]: Well ok.

Comment (K1rakishou) [reply Kuroba/app/src/main/java/com/github/k1rakishou/chan/core/manager/ReplyManager.java]: Those are different things in Java and Kotlin. In Java "==" is reference comparison which only compares pointers to objects while "equals" compares contents of the objects. In Kotlin "==" does the same "equals" does in Java, and "===" does what "==" does in Java.

File: Kuroba/app/build.gradle [hunk 7]
```diff
@@ -290,6 +290,7 @@ dependencies {
     implementation "androidx.appcompat:appcompat:${versions.appcompat}"
     implementation 'androidx.browser:browser:1.2.0'
     implementation 'androidx.constraintlayout:constraintlayout:2.0.1'
+    implementation 'androidx.preference:preference:1.1.1'
```
Comment (K1rakishou) [inline Kuroba/app/build.gradle:293]: Why did you add this dependency, btw?

**Comment by K1rakishou** [comment]
Oh yeah, after the initial review please consider not using rebase because it will replace everything and it will make really hard to track the changes between the reviews. Just add new commits and merge with develop if necessary.

File: Kuroba/app/build.gradle (hunk 7 above)
Comment (TacoTheDank) [reply Kuroba/app/build.gradle:293]: It was needed to fix the Preference deprecation in the second commit.

Comment (K1rakishou) [reply Kuroba/app/build.gradle:293]: Ohh I see.

**Review by K1rakishou — APPROVED** [review]

**Comment by K1rakishou** [comment]
Alright , looks good.
